source/vibe/web/internal/rest/jsclient.d

```diff
@@ -31,85 +59,94 @@ import std.conv : to;
 
 	auto intf = RestInterface!TImpl(settings, true);
 
-	output.formattedWrite("%s = new function() {\n", name.length ? name : intf.I.stringof);
+	auto fout = indentSink(output, jsgenset.indentStep);
 
-	output.put("var toRestString = function(v) { return v; }\n");
+	fout.formattedWrite("%s%s = new function() {\n", jsgenset.parent ? "" : "this.",
+			jsgenset.name.length ? jsgenset.name : intf.I.stringof);
+
+	if (jsgenset.parent)
+		fout.put("var toRestString = function(v) { return JSON.stringify(v); }\n");
 
 	foreach (i, SI; intf.SubInterfaceTypes) {
-		output.generateInterface!SI(__traits(identifier, intf.SubInterfaceFunctions[i]), intf.subInterfaces[i].settings);
+		fout.put("\n");
+		auto childjsset = jsgenset.child(__traits(identifier, intf.SubInterfaceFunctions[i]));
+		fout.generateInterface!SI(intf.subInterfaces[i].settings, childjsset);
 	}
 
 	foreach (i, F; intf.RouteFunctions) {
 		alias FT = FunctionTypeOf!F;
 		auto route = intf.routes[i];
 
 		// function signature
-		output.formattedWrite("  this.%s = function(", route.functionName);
+		fout.put("\n");
+		fout.formattedWrite("this.%s = function(", route.functionName);
 		foreach (j, param; route.parameters) {
-			output.put(param.name);
-			output.put(", ");
+			fout.put(param.name);
+			fout.put(", ");
 		}
-		static if (!is(ReturnType!FT == void)) output.put("on_result, ");
-		output.put("on_error) {\n");
+		static if (!is(ReturnType!FT == void)) fout.put("on_result, ");
+
+		fout.put("on_error) {\n");
 
 		// url assembly
+		fout.put("var url = ");
 		if (route.pathHasPlaceholders) {
 			// extract the server part of the URL
-			output.put("    var url = ");
 			auto burl = URL(intf.baseURL);
 			burl.pathString = "/";
-			output.serializeToJson(burl.toString()[0 .. $-1]);
+			fout.serializeToJson(burl.toString()[0 .. $-1]);
 			// and then assemble the full path piece-wise
 			foreach (p; route.fullPathParts) {
-				output.put(" + ");
-				if (!p.isParameter) output.serializeToJson(p.text);
-				else output.formattedWrite("encodeURIComponent(toRestString(%s))", p.text);
+				fout.put(" + ");
+				if (!p.isParameter) fout.serializeToJson(p.text);
+				else fout.formattedWrite("encodeURIComponent(%s)", p.text);
```

Comment:
I implement this function as [JSON.stringify](https://github.com/rejectedsoftware/vibe.d/pull/1566/files/5745ca3a0af6dbd5e56b3dd08619850f5e69de19#diff-00965ba02c942ce6f16f4106fcd28900R68), because if you want use objects as a parameter of get request you get `object.Object` in params instead `{"x":2, "y":3}` for example. If use it in this case result url has quotes symbols. Maybe need add `toGetParameter` for get request?